### PR TITLE
Assignment 4

### PR DESCRIPTION
**Pull request by wpmcgrath95**

### CODE REVIEW

File: SQL_files/baseball_batting_average.sql [hunk 1]
```diff
@@ -24,7 +24,7 @@ Table: annual table (BA annually)
 */
 DROP TABLE IF EXISTS BATTERS_ANNUALLY;
 CREATE TABLE BATTERS_ANNUALLY AS
-(SELECT baseball.batter_counts.batter AS Batter, REPLACE(YEAR(baseball.game.local_date),',','') AS The_Year,
```
Comment (gramadin) [inline SQL_files/baseball_batting_average.sql:27]: just removing the empty space you found for an old fire? or are you using the baseball stuff in this assignment?

File: requirements.in [hunk 2]
```diff
@@ -2,4 +2,5 @@ numpy
 plotly
 pandas
 sklearn
-pyspark
\ No newline at end of file
+pyspark
+statsmodels
```
Comment (gramadin) [inline requirements.in:6]: add a new line?

File: python_files/ranking_algos.py [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+import pandas as pd
+import statsmodels.api
+from plotly import express as px
+from plotly import figure_factory as ff
+from plotly import graph_objects as go
+from sklearn import datasets
+from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
+from sklearn.metrics import confusion_matrix
+from sklearn.model_selection import train_test_split
+from sklearn.utils.multiclass import type_of_target as tot
```
Comment (gramadin) [inline python_files/ranking_algos.py:32]: name convention is common sense, but just FYI it is the same one i used.

File: plots/AGE_distr_cat_resp_plot.html [hunk 4]
```diff
@@ -0,0 +1,4064 @@
+<html>
```
Comment (gramadin) [inline plots/AGE_distr_cat_resp_plot.html:1]: Did you want to upload all these HTML files?

**Review by gramadin — COMMENTED** [review]
Not sure if you wanted to upload all the HTML files.

File: plots/AGE_distr_cat_resp_plot.html (hunk 4 above)
Comment (gramadin) [reply plots/AGE_distr_cat_resp_plot.html:1]: also not sure if it is specific to me but i get this error when i run your code

Comment (gramadin) [reply plots/AGE_distr_cat_resp_plot.html:1]: try ./ at the beginning of line 141 ??

File: python_files/ranking_algos.py [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+        title_text = "Continuous Response by Continuous Predictor"
+        fig = px.scatter(x=feat, y=y, trendline="ols")
+        fig.update_layout(
+            title=f"{title_text}: {stat_text}",
+            xaxis_title=f"Predictor: {feat.name}",
+            yaxis_title=f"Response: {y_name}",
+        )
+        fig.show()
+        fig.write_html(
+            file=f"plots/{feat.name}_scatter_plot.html", include_plotlyjs="cdn"
```
Comment (gramadin) [inline python_files/ranking_algos.py:143]: add ./ for relative path other wise i get an error when running??

File: python_files/ranking_algos.py [hunk 6]
```diff
@@ -0,0 +1,365 @@
...
+        title_text = "Continuous Response by Categorical Predictor"
+        fig_1 = ff.create_distplot(group_list, group_labels, bin_size=0.2)
+        fig_1.update_layout(
+            title=f"{title_text}: {stat_text}",
+            xaxis_title=f"Response: {y_name}",
+            yaxis_title="Distribution",
+        )
+        fig_1.show()
+        fig_1.write_html(
+            file=f"plots/{feat.name}_distr_cont_resp_plot.html", include_plotlyjs="cdn"
```
Comment (gramadin) [inline python_files/ranking_algos.py:176]: same here

File: python_files/ranking_algos.py [hunk 7]
```diff
@@ -0,0 +1,365 @@
...
+                )
+            )
+        fig_2.update_layout(
+            title=f"{title_text}: {stat_text}",
+            xaxis_title="Groupings",
+            yaxis_title=f"Response: {y_name}",
+        )
+        fig_2.show()
+        fig_2.write_html(
+            file=f"plots/{feat.name}_violin_cont_resp_plot.html", include_plotlyjs="cdn"
```
Comment (gramadin) [inline python_files/ranking_algos.py:197]: same here

File: python_files/ranking_algos.py [hunk 8]
```diff
@@ -0,0 +1,365 @@
...
+        title_text = "Continuous Predictor by Categorical Response"
+        fig_1 = ff.create_distplot(group_list, group_labels, bin_size=2)
+        fig_1.update_layout(
+            title=f"{title_text}: {stat_text}",
+            xaxis_title=f"Predictor: {feat.name}",
+            yaxis_title="Distribution",
+        )
+        fig_1.show()
+        fig_1.write_html(
+            file=f"plots/{feat.name}_distr_cat_resp_plot.html", include_plotlyjs="cdn"
```
Comment (gramadin) [inline python_files/ranking_algos.py:229]: same here

File: python_files/ranking_algos.py [hunk 9]
```diff
@@ -0,0 +1,365 @@
...
+                )
+            )
+        fig_2.update_layout(
+            title=f"{title_text}: {stat_text}",
+            xaxis_title=f"Response: {y_name}",
+            yaxis_title=f"Predictor: {feat.name}",
+        )
+        fig_2.show()
+        fig_2.write_html(
+            file=f"plots/{feat.name}_violin_cat_resp_plot.html", include_plotlyjs="cdn"
```
Comment (gramadin) [inline python_files/ranking_algos.py:250]: same here

File: python_files/ranking_algos.py [hunk 10]
```diff
@@ -0,0 +1,365 @@
...
+
+        fig = go.Figure(data=go.Heatmap(z=conf_matrix, zmin=0, zmax=conf_matrix.max()))
+        fig.update_layout(
+            title=f"{title_text}: {stat_text}",
+            xaxis_title=f"Response: {y_name}",
+            yaxis_title=f"Predictor: {feat.name}",
+        )
+        fig.show()
+        fig.write_html(
+            file=f"plots/{feat.name}_heatmap_cat_resp_plot.html", include_plotlyjs="cdn"
```
Comment (gramadin) [inline python_files/ranking_algos.py:270]: same here

**Review by gramadin — CHANGES_REQUESTED** [review]
added the recommendations for file creation of plots

File: python_files/ranking_algos.py (hunk 5 above)
Comment (gramadin) [reply python_files/ranking_algos.py:143]: FileNotFoundError: [Errno 2] No such file or directory:

Comment (gramadin) [reply python_files/ranking_algos.py:143]: or absolute path

Comment (gramadin) [reply python_files/ranking_algos.py:143]: or import os

Comment (wpmcgrath95) [reply python_files/ranking_algos.py]: ok fixed it thank you!

File: plots/AGE_distr_cat_resp_plot.html (hunk 4 above)
Comment (wpmcgrath95) [reply plots/AGE_distr_cat_resp_plot.html]: no I didn't good catch!

File: python_files/ranking_algos.py (hunk 6 above)
Comment (wpmcgrath95) [reply python_files/ranking_algos.py]: ok thank you. Fixed it in the update

**Review by dafrenchyman — APPROVED** [review]
Approved. Please merge in when you get a chance. Please review comments in canvas for more details about my review.

**Pull request merged**